### PR TITLE
mambaforge: Add version 4.12.0-0

### PR DESCRIPTION
**Pull request by ivaquero**
add mambaforge-4.12.0-0

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/mambaforge.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+    "version": "4.12.0-0",
+    "description": "A conda-forge distribution",
+    "homepage": "https://mirrors.tuna.tsinghua.edu.cn/github-release/conda-forge/miniforge/",
```
Comment (issaclin32) [inline bucket/mambaforge.json:4]:
```suggestion
    "homepage": "https://github.com/conda-forge/miniforge",
```

File: bucket/mambaforge.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            "python.exe",
+            "python3"
+        ]
+    ],
+    "env_add_path": [
+        "scripts",
+        "Library\\bin"
+    ],
+    "persist": "envs",
+    "checkver": {
```
Comment (issaclin32) [inline bucket/mambaforge.json:43]:
```suggestion
    "checkver": "github",
```

File: bucket/mambaforge.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        "github": "https://github.com/conda-forge/miniforge"
```
Comment (issaclin32) [inline bucket/mambaforge.json:44]:
```suggestion
```

File: bucket/mambaforge.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    },
```
Comment (issaclin32) [inline bucket/mambaforge.json:45]:
```suggestion
```

File: bucket/mambaforge.json (hunk 3 above)
Comment (issaclin32) [inline bucket/mambaforge.json:44]:
```suggestion
        "github": "https://github.com/conda-forge/miniforge",
        "regex": "tag/([\\d.-]+)"
```

File: bucket/mambaforge.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://github.com/conda-forge/miniforge/releases/download/$version/Mambaforge-$version-Windows-x86_64.exe",
+                "hash": {
+                    "url": "$baseurl",
```
Comment (issaclin32) [inline bucket/mambaforge.json:51]:
```suggestion
                    "url": "$url.sha256"
```

File: bucket/mambaforge.json [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                    "regex": "(?sm)$basename<.*?<td>$md5</td>"
```
Comment (issaclin32) [inline bucket/mambaforge.json:52]:
```suggestion
```

File: bucket/mambaforge.json (hunk 3 above)
Comment (issaclin32) [reply bucket/mambaforge.json:44]: `regex` is needed because Scoop matches `tag/[\\d.]+` by default. The version 4.12.0**-0** cannot be correctly matched.

**Comment by rashil2000** [comment]
What about extracting the installer instead of running it?

**Comment by issaclin32** [comment]
> What about extracting the installer instead of running it?

I tried. However it could not be extracted by *7zip*, *innounp*, *isx*, or even *uniextract2*.

**Review by issaclin32 — APPROVED** [review]

**Comment by ivaquero** [comment]
> /verify

Thanks again

**Review by rashil2000 — APPROVED** [review]

**Pull request merged**